lib/discord_bot/discord_bot.ex

```diff
@@ -0,0 +1,188 @@
+defmodule DiscordSupervisor do
+  use Supervisor
+  alias Alchemy.Client
+  alias Alchemy.Embed
+  require Logger
+
+  defmodule DiscordBot do
+    use GenServer
+
+    defmodule HubsCommands do
+      use Alchemy.Cogs
+      require Alchemy.Embed, as: Embed
+
+      Cogs.def duck do
+        Cogs.say("Quack :duck:")
+      end
+
+      Cogs.def bound do
+        case Cachex.get!(:discord_bot_state, "hubs_for_channel")[message.channel_id] do
+          hub_ids = [_ | _] ->
+            Cogs.say("This channel is bound to #{Enum.join(hub_ids, ", ")}")
+
+          _ ->
+            Cogs.say("No hubs bound to this channel")
+        end
+      end
+    end
+
+    defmodule HubsEvents do
+      use Alchemy.Events
+
+      Events.on_channel_update(:on_channel_update)
+      Events.on_ready(:on_ready)
+      Events.on_message(:on_message)
+
+      def on_channel_update(_channel) do
+        update_bound_channels()
+      end
+
+      def on_ready(_shard, _total_shards) do
+        update_bound_channels()
+      end
+
+      def on_message(msg) do
+        if hub_ids = !msg.author.bot && Cachex.get!(:discord_bot_state, "hubs_for_channel")[msg.channel_id] do
+          Logger.debug "Outgoing message from Discord to Hubs: #{msg.content}"
+          hub_ids |> Enum.each(&broadcast_to_hubs(&1, msg))
+        end
+      end
+
+      def broadcast_to_hubs(hub_id, msg) do
+        RetWeb.Endpoint.broadcast!("hub:#{hub_id}", "message", %{
+              type: "chat",
+              body: msg.content,
+              from: "#{msg.author.username}##{msg.author.discriminator}"
+                                   })
+      end
+
+      def update_bound_channels do
+        hostnames = Application.get_env(:ret, Elixir.DiscordBot)[:hostnames] |> String.split
```

Comment:
I'd suggest we fall through to using `RetWeb.Endpoint.host` if the config is not set, and then just don't bother setting it in prod (unless you think we can get away without `localhost` in dev, in which case this config isn't needed at all then I think.)

Reply:
also there's a pattern I started using for grabbing module configs: 

https://github.com/mozilla/reticulum/blob/master/lib/ret/media_resolver.ex#L349

and then using with to handle case where not defined:

https://github.com/mozilla/reticulum/blob/master/lib/ret/media_resolver.ex#L329

Reply:
I prefer setting this explicitly, because each environment is associated with multiple hostnames. For example, production is associated with both hubs.mozilla.com and smoke-hubs.mozilla.com.

Reply:
> also there's a pattern I started using for grabbing module configs:

I explicitly did not use this pattern because the Discord configuration is used by two different modules -- `DiscordSupervisor` and `DiscordBot`, and I didn't want to split it into separate configuration sections.

Reply:
Ah that makes sense, because we share the cluster with smoke `RetWeb.Endpoint.host` varies intra-cluster, which seems fairly problematic now that we're trying to do fancier things like this.

Re: configs, it's probably OK either way, I just wanted to point out the other pattern. I think using a different module's name for the configs is not a big deal.